group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -619,6 +622,16 @@ ConsumerGroup getOrMaybeCreateConsumerGroup(
         }
     }
 
+    /**
+     * An overloaded method of {@link GroupMetadataManager#getOrMaybeCreateConsumerGroup(String, boolean, List<Record>)}
+     */
+    ConsumerGroup getOrMaybeCreateConsumerGroup(
```

Comment:
It looks like that this one is always used with `createIfNotExists` set to `false`. Should we remove the flag and rename the method?